src/rules/no-ignored-return.ts

```diff
@@ -225,6 +224,9 @@ function isReplaceWithCallback(
   if (methodName === 'replace' && callArguments.length > 1) {
     const type = getTypeFromTreeNode(callArguments[1], services);
     const typeNode = services.program.getTypeChecker().typeToTypeNode(type, undefined, undefined);
+    // dynamically import 'typescript' as classic 'import' will fail if project not using 'typescript' parser
+    // we are sure it's available as 'RequiredParserServices' are available here
+    const ts = require('typescript');
```

Comment:
I would suggest writing a unit test with an internal rule that would raise issues on static import statements importing TypeScript dependency. This way, we would remember dynamically importing it when needed.

Reply:
I didn't get what are you suggesting, a unit test checking what?

Reply:
Unit test checking the following pattern in a rule implementation:

```typescript
import * as ts from 'typescript'
```

and suggesting to rewrite it as:

```typescript
const ts = require('typescript');
```

while emphasising to move it were it is actually used.

This unit test might become useless depending on what we are going to do with TypeScript dependency: dependency, peerDependency, etc.

Reply:
I will write custom rule in another PR